master/buildbot/steps/source/repo.py

```diff
@@ -394,7 +403,12 @@ def computeTarballOptions(self):
         # can be non negligible
         tar = ['tar']
         if self.tarball.endswith("gz"):
-            tar.append('-z')
+            # pigz is a parallel gzip that is much faster on multicore machines
+            if executable_in_path('pigz'):
```

Comment:
I think it is more reasonable to activate this feature by asking user to end the tarball name with .pigz